src/ScriptEditor/ui/ScriptEditorRoot.tsx

```diff
@@ -211,18 +216,28 @@ function Root(props: IProps): React.ReactElement {
 
   useEffect(() => {
     function keydown(event: KeyboardEvent): void {
-      if (Settings.DisableHotkeys) return;
-      //Ctrl + b
-      if (event.code == "KeyB" && (event.ctrlKey || event.metaKey)) {
-        event.preventDefault();
-        Router.toPage(Page.Terminal);
+      if (Settings.DisableHotkeys) {
+        return;
       }
-
-      // CTRL/CMD + S
-      if (event.code == "KeyS" && (event.ctrlKey || event.metaKey)) {
-        event.preventDefault();
-        event.stopPropagation();
-        save();
+      if (event.getModifierState(event.key)) {
+        return;
+      }
+      const keyBindingTypes = determineKeyBindingTypes(
+        Settings.KeyBindings,
+        convertKeyboardEventToKeyCombination(event),
+      );
+      for (const keyBindingType of keyBindingTypes) {
```

Comment:
Since you have these as a set, you can do `if keyBindingTypes.has()` blocks instead of a loop+switch. Slightly shorter and also more efficient. It also guarantees that overlap between save+gototerminal will be handled in the order we want.